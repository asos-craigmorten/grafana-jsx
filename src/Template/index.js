import PropTypes from "prop-types";

const Template = ({
  allFormat = null,
  allValue = null,
  current,
  datasource,
  definition,
  hide = 0,
  includeAll = false,
  label = null,
  multi = false,
  multiFormat,
  name,
  options = [],
  query,
  refresh = 0,
  regex,
  skipUrlSync = false,
  sort = 0,
  tagValuesQuery,
  tags = [],
  tagsQuery,
  type,
  useTags = false,
  ...props
}) => (
  <template
    allFormat={allFormat}
    allValue={allValue}
    current={current}
    datasource={datasource}
    definition={definition}
    hide={hide}
    includeAll={includeAll}
    label={label}
    multi={multi}
    multiFormat={multiFormat}
    name={name}
    options={options}
    query={query}
    refresh={refresh}
    regex={regex}
    skipUrlSync={skipUrlSync}
    sort={sort}
    tags={tags}
    tagsQuery={tagsQuery}
    tagValuesQuery={tagValuesQuery}
    type={type}
    useTags={useTags}
    {...props}
  />
);

Template.PropTypes = {
  allFormat: PropTypes.string,
  allValue: PropTypes.any,
  current: PropTypes.shape({
    isNone: PropTypes.bool,
    selected: PropTypes.bool,
    text: PropTypes.string,
    value: PropTypes.string,
  }),
  datasource: PropTypes.string,
  definition: PropTypes.string,
  hide: PropTypes.oneOf([0, 1, 2]),
  includeAll: PropTypes.bool,
  label: PropTypes.string,
  multi: PropTypes.bool,
  multiFormat: PropTypes.any,
  name: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.any),
  query: PropTypes.string,
  refresh: PropTypes.oneOf([0, 1, 2]),
  regex: PropTypes.string,
  skipUrlSync: PropTypes.bool,
  sort: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 6]),
  tagValuesQuery: PropTypes.string,
  tags: PropTypes.arrayOf(PropTypes.string),
  tagsQuery: PropTypes.string,
  type: PropTypes.string,
  useTags: PropTypes.bool,
};

export default Template;
