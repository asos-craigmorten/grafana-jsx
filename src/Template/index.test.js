import Template from "./";

const withAllProps = {
  allFormat: "test-all-format",
  allValue: "test-all-value",
  current: {
    selected: true,
    text: "test-current-text",
    value: "test-current-value",
  },
  datasource: "test-datasource",
  definition: "test-definition",
  hide: 0,
  includeAll: true,
  label: "test-label",
  multi: true,
  multiFormat: "test-multi-format",
  name: "test-name",
  options: ["test-option-1", "test-option-2"],
  query: "test-query",
  refresh: 0,
  regex: "test-regex",
  skipUrlSync: true,
  sort: 0,
  tagValuesQuery: "test-tag-values-query",
  tags: ["test-tag-1", "test-tag-2"],
  tagsQuery: "test-tags-query",
  type: "test-type",
  useTags: true,
};

describe("Template", () => {
  describe("when passed no props", () => {
    it("should render a default Template", () => {
      expect(<Template />).toMatchSnapshot();
    });
  });

  describe("when passed all props", () => {
    it("should render a custom Template", () => {
      expect(<Template {...withAllProps} />).toMatchSnapshot();
    });
  });

  describe("when passed additional props", () => {
    it("should render a custom Template with the additional props", () => {
      expect(
        <Template
          {...withAllProps}
          test-template-prop={"test-template-value"}
        />
      ).toMatchSnapshot();
    });
  });
});
