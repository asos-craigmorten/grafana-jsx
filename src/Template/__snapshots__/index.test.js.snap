// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template when passed additional props should render a custom Template with the additional props 1`] = `
Object {
  "allFormat": "test-all-format",
  "allValue": "test-all-value",
  "current": Object {
    "selected": true,
    "text": "test-current-text",
    "value": "test-current-value",
  },
  "datasource": "test-datasource",
  "definition": "test-definition",
  "hide": 0,
  "includeAll": true,
  "label": "test-label",
  "multi": true,
  "multiFormat": "test-multi-format",
  "name": "test-name",
  "options": Array [
    "test-option-1",
    "test-option-2",
  ],
  "query": "test-query",
  "refresh": 0,
  "regex": "test-regex",
  "skipUrlSync": true,
  "sort": 0,
  "tagValuesQuery": "test-tag-values-query",
  "tags": Array [
    "test-tag-1",
    "test-tag-2",
  ],
  "tagsQuery": "test-tags-query",
  "test-template-prop": "test-template-value",
  "type": "test-type",
  "useTags": true,
}
`;

exports[`Template when passed all props should render a custom Template 1`] = `
Object {
  "allFormat": "test-all-format",
  "allValue": "test-all-value",
  "current": Object {
    "selected": true,
    "text": "test-current-text",
    "value": "test-current-value",
  },
  "datasource": "test-datasource",
  "definition": "test-definition",
  "hide": 0,
  "includeAll": true,
  "label": "test-label",
  "multi": true,
  "multiFormat": "test-multi-format",
  "name": "test-name",
  "options": Array [
    "test-option-1",
    "test-option-2",
  ],
  "query": "test-query",
  "refresh": 0,
  "regex": "test-regex",
  "skipUrlSync": true,
  "sort": 0,
  "tagValuesQuery": "test-tag-values-query",
  "tags": Array [
    "test-tag-1",
    "test-tag-2",
  ],
  "tagsQuery": "test-tags-query",
  "type": "test-type",
  "useTags": true,
}
`;

exports[`Template when passed no props should render a default Template 1`] = `
Object {
  "allFormat": null,
  "allValue": null,
  "hide": 0,
  "includeAll": false,
  "label": null,
  "multi": false,
  "options": Array [],
  "refresh": 0,
  "skipUrlSync": false,
  "sort": 0,
  "tags": Array [],
  "useTags": false,
}
`;
